packages/backend/src/auto-peering/errors.ts

```diff
@@ -0,0 +1,38 @@
+export enum AutoPeeringError {
+  InvalidIlpConfiguration = 'InvalidIlpConfiguration',
+  InvalidPeerIlpConfiguration = 'InvalidPeerIlpConfiguration',
+  UnknownAsset = 'UnknownAsset',
+  PeerUnsupportedAsset = 'PeerUnsupportedAsset',
+  InvalidPeerUrl = 'InvalidPeerUrl',
+  InvalidPeeringRequest = 'InvalidPeeringRequest',
+  DuplicatePeer = 'DuplicatePeer'
+}
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any, @typescript-eslint/explicit-module-boundary-types
+export const isAutoPeeringError = (o: any): o is AutoPeeringError =>
+  Object.values(AutoPeeringError).includes(o)
+
+export const errorToCode: {
+  [key in AutoPeeringError]: number
+} = {
+  [AutoPeeringError.InvalidIlpConfiguration]: 400,
+  [AutoPeeringError.InvalidPeerIlpConfiguration]: 400,
+  [AutoPeeringError.UnknownAsset]: 404,
+  [AutoPeeringError.PeerUnsupportedAsset]: 400,
+  [AutoPeeringError.InvalidPeerUrl]: 400,
+  [AutoPeeringError.InvalidPeeringRequest]: 400,
+  [AutoPeeringError.DuplicatePeer]: 400
+}
+
+export const errorToMessage: {
+  [key in AutoPeeringError]: string
+} = {
+  [AutoPeeringError.InvalidIlpConfiguration]:
+    'The ILP configuration is misconfigured',
+  [AutoPeeringError.InvalidPeerIlpConfiguration]: `Requested peer's ILP configuration is misconfigured`,
+  [AutoPeeringError.UnknownAsset]: 'Unknown asset',
+  [AutoPeeringError.PeerUnsupportedAsset]: 'Peer does not support asset',
```

Comment:
How does the error related to unsupported assets work?
- when one of the provided assets is not supported it rollbacks everything
- when one of the provided assets is not supported it keeps creating the others but will return the message `AutoPeeringError.PeerUnsupportedAsset`

And what is the difference between `AutoPeeringError.PeerUnsupportedAsset` and `AutoPeeringError.UnknownAsset`

Reply:
Originally, I was thinking about changing how we define the "Peer" in Rafiki to be relationship to be 1 Peer many assets, but after consideration w Sabine we decided it's better to keep what we have now 1 "Peer" defines an ILP relationship over a single asset: it keeps liquidity management a little bit easier to understand, and doesn't require a big overhaul in the service/DB. 

As a result, 
> when one of the provided assets is not supported
this isn't a scenario that would happen, given we only try to attempt to auto-peer using one asset

Reply:
> And what is the difference between AutoPeeringError.PeerUnsupportedAsset and AutoPeeringError.UnknownAsset

from the perspective of peer 1 trying to peer with peer 2, AutoPeeringError.UnknownAsset
 is if your own (peer 1) instance does not support the asset and AutoPeeringError.PeerUnsupportedAsset is when trying to peer with peer 2, peer 2 does not support the asset

Reply:
Yes, it's just the case for the wrong input